js/models/message_handler.js

```diff
@@ -29,14 +29,56 @@ export default class MessageHandler {
     this.aladin.addMOC(A.MOCFromJSON(msg["moc_dict"], options));
   }
 
-  handleAddOverlayFromSTCS(msg) {
-    const overlayOptions = convertOptionNamesToCamelCase(
-      msg["overlay_options"] || {},
+  handleAddOverlay(msg) {
+    const regions = msg["regions_infos"];
+    const graphic_options = convertOptionNamesToCamelCase(
+      msg["graphic_options"] || {},
     );
-    const stcString = msg["stc_string"];
-    const overlay = A.graphicOverlay(overlayOptions);
+    if (!graphic_options["color"]) graphic_options["color"] = "red";
+    const overlay = A.graphicOverlay(graphic_options);
     this.aladin.addOverlay(overlay);
-    overlay.addFootprints(A.footprintsFromSTCS(stcString));
+    for (const region of regions) {
+      const infos = region["infos"];
+      switch (region["region_type"]) {
+        case "stcs":
+          overlay.addFootprints(
+            A.footprintsFromSTCS(infos.stcs, region.options),
+          );
+          break;
+        case "circle":
+          overlay.add(
+            A.circle(infos.ra, infos.dec, infos.radius, region.options),
+          );
+          break;
+        case "ellipse":
+          overlay.add(
+            A.ellipse(
+              infos.ra,
+              infos.dec,
+              infos.a,
+              infos.b,
+              infos.theta,
+              region.options,
+            ),
+          );
+          break;
+        case "line":
+          overlay.add(
+            A.line(
+              infos.ra1,
+              infos.dec1,
+              infos.ra2,
+              infos.dec2,
+              "ICRS",
```

Comment:
Aladin Lite js API here is not consistent. I will change the optional frame param to put it as a key into the regular option object following. So I would recommend you to remove the ICRS. I created the issue in aladin lite: https://github.com/cds-astro/aladin-lite/issues/156

Reply:
I let the "ICRS" param instead of null to be sure that the line use ICRS coordinates and not the current frame ones.

Reply:
Ok I see. Can you put it in the options object ? That is also relevant for the other footprints as they all call world2pix inside i.e. not giving a frame will consider the frame as the current one.